app/controllers/supermarkets_controller.rb

```diff
@@ -0,0 +1,6 @@
+class SupermarketsController < ApplicationController
+  def show
+    @supermarket = Supermarket.find(params[:id])
+    @customers = @supermarket.customers.distinct
```

Comment:
We will want to get our unique list of customers at the model level. So you will want to create a method in your `supermarket` model and put `customers.distinct` in there and then call that method in your view.